utils/src/main/kotlin/Extensions.kt

```diff
@@ -232,7 +232,13 @@ inline fun <K, V, W> Map<K, V>.zipWithDefault(other: Map<K, V>, default: V, oper
 /**
  * Return the string encoded for safe use as a file name or "unknown", if the string is empty.
  */
-fun String.encodeOrUnknown() = fileSystemEncode().takeUnless { it.isEmpty() } ?: "unknown"
+fun String.encodeOrUnknown(): String = encodeOr("unknown")
+
+/**
+ * Return the string encoded for safe use as a file name or [default] encoded for safe use as a file name, if this
+ * string is empty.
+ */
+fun String.encodeOr(default: String): String = ifEmpty { default }.fileSystemEncode()
```

Comment:
Note that this is a change in behavior compared to before: Previously, the default string (i.e. "unknown") was taken literally and not encoded. At the example of "unknown" it does not matter, and probably also not for your use-case, but I still think it's a bit misleading as the function name is "encodeOr" which sounds like "encode it or take whatever I provide (without encoding it)".

I'm not saying we should change the implementation, as I actually like to be on the safe side and also encode the `default` string. Also the docs are pretty clear on the implemented behavior, and I cannot think of a better function name right now.

So bottom line, maybe it's enough to just document the behavior change and why it was done / should not matter in most cases in the commit message.

Reply:
Oh, and can we call `default` also `emptyValue` like you do in the next commit?